soap/soap_test.go

```diff
@@ -0,0 +1,117 @@
+package soap
+
+import (
+	"encoding/xml"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type Ping struct {
+	XMLName xml.Name `xml:"http://example.com/service.xsd Ping"`
+
+	Request *PingRequest `xml:"request,omitempty"`
+}
+
+type PingRequest struct {
+	// XMLName xml.Name `xml:"http://example.com/service.xsd PingRequest"`
```

Comment:
There are some problems with xml unmarshaling. If I uncomment the following XMLName I get error:
```
xml: name "request" in tag of soap.Ping.Request conflicts with name "PingRequest" in *soap.PingRequest.XMLName
```
@c4milo Do you know what problem can it be? To me it looks like a xml package problem with namespaces.

Reply:
Is it similar to this issue? https://github.com/hooklift/gowsdl/issues/17 If so, yes.

Reply:
Ok, I see, it sound similar. Thanks 👍